packages/kit/src/exports/vite/index.js

```diff
@@ -369,12 +369,27 @@ async function kit({ svelte_config }) {
 
 				if (!secondary_build_started) {
 					manifest_data = sync.all(svelte_config, config_env.mode).manifest_data;
+					// During the initial server build we don't know yet
+					new_config.define.__SVELTEKIT_HAS_SERVER_LOAD__ = 'true';
+					new_config.define.__SVELTEKIT_HAS_UNIVERSAL_LOAD__ = 'true';
+				} else {
+					// Through the finished analysis we can now check if any node has server or universal load functions
+					const has_server_load = build_metadata
+						? Object.values(build_metadata.nodes).some((node) => node.has_server_load)
+						: true;
+					const has_universal_load = build_metadata
+						? Object.values(build_metadata.nodes).some((node) => node.has_universal_load)
+						: true;
+					new_config.define.__SVELTEKIT_HAS_SERVER_LOAD__ = s(has_server_load);
+					new_config.define.__SVELTEKIT_HAS_UNIVERSAL_LOAD__ = s(has_universal_load);
 				}
```

Comment:
`build_metadata` is guaranteed to exist at this point, so rather than having the pair of ternaries (which make it seem like there are more possibilities than there actually are) what if we do this, using `build_metadata` as a proxy for `secondary_build_started`?

```suggestion
				if (build_metadata) {
					const nodes = Object.values(build_metadata.nodes);

					// Through the finished analysis we can now check if any node has server or universal load functions
					const has_server_load = nodes.some((node) => node.has_server_load);
					const has_universal_load = nodes.some((node) => node.has_universal_load);

					new_config.define.__SVELTEKIT_HAS_SERVER_LOAD__ = s(has_server_load);
					new_config.define.__SVELTEKIT_HAS_UNIVERSAL_LOAD__ = s(has_universal_load);
				} else {
					manifest_data = sync.all(svelte_config, config_env.mode).manifest_data;
					
					// During the initial server build we don't know yet
					new_config.define.__SVELTEKIT_HAS_SERVER_LOAD__ = 'true';
					new_config.define.__SVELTEKIT_HAS_UNIVERSAL_LOAD__ = 'true';
				}
```

Reply:
I like the idea of making it not a ternary but I'd prefer a type cast instead (that will also fail loudly if something goes wrong down the line)